### PR TITLE
Test data in sqlite db

### PR DESCRIPTION
**Pull request by vvscode**
@ewnd9  it's instead of commented part of tests. Possible it worth to add more specific tests for db structure on the output. But I believe that we need it if I'll add work with tags / something else. 

It's not big step. But hope it makes test a little more valuable

### CODE REVIEW

**Comment by vvscode** [comment]
Also I'm interested about target versions - does it make sense to support node 0.10 / 0.12 ? ))

File: test/test.js [hunk 1]
```diff
@@ -1,13 +1,26 @@
+"use strict";
```
Comment (ewnd9) [inline test/test.js]: Babel inserting it by default, see https://babeljs.io/repl/

File: test/test.js [hunk 2]
```diff
@@ -1,13 +1,26 @@
...
 import test from 'ava';
 
 import 'babel-register';
 import 'babel-polyfill';
 
 import AnkiExport from '../src/index';
 import fs from 'fs';
+import  decompress from 'decompress';
```
Comment (ewnd9) [inline test/test.js]: Please remove white-space before `decompress`

File: test/test.js [hunk 3]
```diff
@@ -1,13 +1,26 @@
...
+import _ from 'lodash';
```
Comment (ewnd9) [inline test/test.js]: Please change to `import sortBy from 'lodash.sortby';`. See below why `isEqual` is not needed

File: test/test.js [hunk 4]
```diff
@@ -1,13 +1,26 @@
...
+import sqlite3 from 'sqlite3';
+import { exec } from  'child_process';
 
-const dest = '/tmp/result.apkg';
-const sample = __dirname + '/fixtures/output.apkg';
+let SEPARATOR = "\u001F";
```
Comment (ewnd9) [inline test/test.js]: Please change double quotes to single and `let` back to `const`

File: test/test.js [hunk 5]
```diff
@@ -22,12 +35,38 @@ test('equals to sample', async t => {
...
+  const normilizedResult = _.sortBy(result.map(({front, back}) => ({
+    front,
+    back: back.split(SEPARATOR).pop()
+  })), 'front');
 
-  // t.ok(destZip === sampleZip);
-  //
-  // i forgot that there are random number involved, so every time slightly different result
-  // one of solutions could be comparing sql data extracted from the apkg archive
+  t.ok(_.isEqual(normilizedResult, cards));
```
Comment (ewnd9) [inline test/test.js]: Please change to `t.deepEqual(normalizedResult, card)`. `ava` uses `lodash#isEqual` internally https://github.com/avajs/ava/blob/82677554efe223fddcfa9f89db69a106611d257e/lib/assert.js#L4

File: test/test.js [hunk 6]
```diff
@@ -22,12 +35,38 @@ test('equals to sample', async t => {
...
+      notes.flds as back
+      from cards JOIN notes where cards.nid = notes.id ORDER BY cards.id`,
+    (err, data) => err ? reject(err) : resolve(data)
+  ));
+  db.close();
 
-  const sampleZip = fs.readFileSync(sample);
-  const destZip = fs.readFileSync(dest);
+  // compare content from just created db with original list of cards
+  const normilizedResult = _.sortBy(result.map(({front, back}) => ({
```
Comment (ewnd9) [inline test/test.js]: `_.sortBy` => `sortBy`

File: test/test.js [hunk 7]
```diff
@@ -22,12 +35,38 @@ test('equals to sample', async t => {
...
+  ];
+  addCards(apkg, cards);
+  const zip = await apkg.save();
+  fs.writeFileSync(dest, zip, 'binary');
+
+  // extract dec to tmp directory
+  await decompress(dest, destUnpacked);
+  // analize db via sqlite
+  const db = new sqlite3.Database(destUnpackedDb);
+  const result = await new Promise((resolve, reject) => db.all(
```
Comment (ewnd9) [inline test/test.js]: `await pify(db.all.bind(db))(`

File: test/test.js [hunk 8]
```diff
@@ -1,13 +1,26 @@
...
+let SEPARATOR = "\u001F";
+let tmpDir = '/tmp/';
+let dest = tmpDir + 'result.apkg';
+let destUnpacked = tmpDir + 'unpacked_result';
+let destUnpackedDb = destUnpacked + '/collection.anki2';
+let sample = __dirname + '/fixtures/output.apkg';
+
+let addCards = (apkg, list) => list.forEach(({front, back}) => apkg.addCard(front, back));
+
+test.beforeEach(async t => new Promise((resolve) => exec(`rm -rf ${dest} ${destUnpacked}`, resolve)));
```
Comment (ewnd9) [inline test/test.js]: Please add the `pify` dependency and change to

```js
await pify(exec)(`rm -rf ${dest} ${destUnpacked}`)
```

File: test/test.js (hunk 1 above)
Comment (vvscode) [inline test/test.js]: it's for `node 0.10.0 / 0.12.0`. When I try to fix tests for this versions. It allow to pass, but it stuck on using dependencies

**Comment by ewnd9** [comment]
> I believe that we need it if I'll add work with tags / something else.

Contributing lacking tests first is definitely a good choice :+1:

> Also I'm interested about target versions - does it make sense to support node 0.10 / 0.12 ? ))
> The question caused support for arrow functions / const / external modules etc   

If you look at [Travis log](https://travis-ci.org/ewnd9/anki-apkg-export/jobs/179244861#L382-L384), you will see the problem is in [`decompress`](https://github.com/kevva/decompress) which dropped `node < 4`.

I am okay with removing `0.10`/`0.12` from CI. I usually treat them as a safeguard for libraries that might be used in browser (e.g. we shouldn't use `decompress` as a runtime dependency because of the old browsers), but according to https://github.com/nodejs/LTS I need to find another way, `0.10` is already not maintained, `0.12` will be in a month

File: test/test.js (hunk 1 above)
Comment (ewnd9) [inline test/test.js]: Is it this error https://travis-ci.org/ewnd9/anki-apkg-export/jobs/179239375#L223 ?

Comment (vvscode) [inline test/test.js]: this one https://travis-ci.org/ewnd9/anki-apkg-export/jobs/179224350#L387

I can downgrade / search another package. But suppose that  the game isn't worth the candle

Comment (ewnd9) [inline test/test.js]: Yep, see my comment below https://github.com/ewnd9/anki-apkg-export/pull/5#issuecomment-263160745

**Comment by vvscode** [comment]
that's why I add `decompress` as dev-dependency. It used only for tests.

I'm ready to wait till the end of the year when node 0.12 won't support more ))

**Comment by ewnd9** [comment]
One last thing, have you tried reusing `jszip` instead of `decompress`?

**Comment by vvscode** [comment]
@ewnd9  yep. `jszip` is too low-level. It's ok if we need create/add files. But unzip tree is not so easy. I spent about an hour - extract flat structure / single file is ok. But thus we have `media` with internal files ( and I think it's not hard to image situation when we'll try check it's content ) - extraction become not so easy

I'm not fan of heavy reliance from single solution. So I just extract logic into helpers layer with async interface (`(pathToZip, pathToUnzip) => promise`)  so it should be easy to change implementation in future

**Comment by ewnd9** [comment]
Squashed and merged :tada: Great work :clap:

**Comment by vvscode** [comment]
Ooops. I just find solution with jszip )

**Pull request merged**